File: metalanguage/src/main/java/entities/MLEntity.java
package entities;

import attributes.MLAttribute;
import constraints.MLConstraint;
import constructs.MLConstruct;
import lombok.Getter;
import lombok.NonNull;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Getter
public class MLEntity extends MLConstruct {
    @Setter
    private long maxCount;
    private List<MLAttribute> attributes;
    private List<MLConstraint> constraints;
    @Setter
    private boolean unique;

    public MLEntity(@NonNull String name, @NonNull UUID id, @NonNull UUID prototypeId) {
        super(name, id, prototypeId);
        attributes = new ArrayList<MLAttribute>();
        constraints = new ArrayList<MLConstraint>();
    }
}

File: metalanguage-lib/src/main/kotlin/ru/psu/entities/MLEntity.kt
package ru.psu.entities

import ru.psu.attributes.MLAttribute
import ru.psu.constraints.MLConstraint
import ru.psu.constructs.MLConstruct
import java.util.*
import kotlin.collections.ArrayList

//Класс сущности в MetaLanguage 1.1
class MLEntity(
        parentId:UUID?, //Идентификатор конструкции-родителя
        id: UUID, //Id сущности
        name: String, //Название
        prototypeId: UUID?, //Id прототипа (нет - сущность в метамодели, есть - экз. сущности)
        innerStructure: UUID?, //Id графа, описывающего внутреннюю структуру
        maxCount:Int //Максимальное число экземпляров
): MLConstruct(parentId, id, name, prototypeId, innerStructure, maxCount) {
    val ports:MutableSet<UUID> = HashSet() //Порты сущности
    val attributes:MutableList<MLAttribute> = ArrayList() //Атрибуты сущности
    val constraints:MutableList<MLConstraint> = ArrayList() //Ограничения на сущность
}